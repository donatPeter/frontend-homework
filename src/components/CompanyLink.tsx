import { Company, ReduxState } from "../types/types";
import styled from "styled-components";
import { connect, useSelector } from "react-redux";
import { getSelectedCompanyId, isCompanySelected } from "../store/selectors";
import { createStructuredSelector } from "reselect";

const StyledCompanyLinkName = styled.div<{ $isSelected: boolean }>`
  padding: 12px 17px;
  font-weight: bold;
  background: ${({ $isSelected }) =>
    $isSelected ? "rgba(55, 164, 71, 0.1)" : "unset"};
`;

interface Props {
  company: Company;
}

type ReduxProps = {
  selectedCompanyId: number | null;
};

const CompanyLink = ({
  company: { name, id },
  selectedCompanyId,
}: Props & ReduxProps) => {
  return (
    <StyledCompanyLinkName $isSelected={selectedCompanyId === id}>
      {name}
      <i />
    </StyledCompanyLinkName>
  );
};

export default connect(
  createStructuredSelector<ReduxState, ReduxProps>({
    selectedCompanyId: getSelectedCompanyId,
  })
)(CompanyLink);

// export default CompanyLink;
