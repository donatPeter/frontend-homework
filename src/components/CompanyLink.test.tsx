import { shallow } from "enzyme";
import { CompanyLink } from "./CompanyLink";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("<CompanyLink />", () => {
  const toggleDropdownMenuVisibilityMock = jest.fn();
  const setSelectedCompanyIdMock = jest.fn();
  const company = { id: 1, name: "Dummy Company" };

  const wrapper = (props: any) =>
    shallow(
      <CompanyLink
        toggleDropdownMenuVisibility={toggleDropdownMenuVisibilityMock}
        setSelectedCompanyId={setSelectedCompanyIdMock}
        company={company}
        {...props}
      />
    );

  it("renders link", () => {
    expect(wrapper({})).toMatchSnapshot();
  });

  it("invokes onClick callback on click", () => {
    const onClick = jest.fn();
    const { getByRole } = render(
      <CompanyLink company={company} onClick={onClick} isSelected={true} />
    );
    expect(onClick).not.toHaveBeenCalled();
    userEvent.click(getByRole("menuitem"));
    expect(onClick).toHaveBeenCalled();
  });
});
