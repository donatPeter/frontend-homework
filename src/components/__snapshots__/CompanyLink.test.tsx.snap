// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyLink /> renders link 1`] = `
<styled.div
  onClick={[Function]}
  role="menuitem"
>
  Dummy Company
</styled.div>
`;
